index.tsx

```diff
@@ -58,10 +59,10 @@ function normalizeBindingNames<Props>(
     return fromPairs(bindingNames.map(_ => [_, '<'])) as AngularBindings<Props>
   }
   if (bindingNames) {
-    return mapValues(bindingNames, _ => _.optional ? '<?' : '<')
+    return mapValues(bindingNames, (_: {optional: boolean}) => _.optional ? '<?' : '<') as AngularBindings<Props>
   }
   if (Class.propTypes) {
-    return mapValues(Class.propTypes, () => '<' as '<')
+    return mapValues(Class.propTypes as {[K in keyof Props]:PropTypes.Requireable<any>}, (_:PropTypes.Requireable<any>) => !_.isRequired ? '<' : '<?') as AngularBindings<Props>
```

Comment:
Shouldn't this be the opposite? If `_.isRequired` then we generate `<`, otherwise we generate `<?`.

Reply:
Yes, this is the most counter intuitive bit, _.isRequired is the prop-types object that defines the required version of the type. If that object exists then the current type is not required. The tests should back this up.